### PR TITLE
Add tasks filters

### LINKED ISSUE #62: Filters tasks functions
Function that filter all relevant tasks according to given key.
The key can be one of:  [created_date, assignee, priority, status]

### PR DESCRIPTION
**Pull request by orzionpour**
Fixes #62 

## Changes you made

- Add tasks filters
- Add tests for tasks filters

### CODE REVIEW

File: tasks/models.py [hunk 1]
```diff
@@ -28,6 +28,29 @@ class Task(models.Model):
     def __str__(self) -> str:
         return self.title
 
+    @classmethod
+    def filterByStatus(cls, status_filter):
```
Comment (kobihk) [inline tasks/models.py:32]: the pythonic way is like:
```suggestion
    def filter_by_status(cls, status_filter):
```

File: tasks/models.py [hunk 2]
```diff
@@ -28,6 +28,29 @@ class Task(models.Model):
...
+        if not isinstance(status_filter, Status):
+            raise ValueError
+        return cls.objects.filter(status=status_filter)
+
+    @classmethod
+    def filterByAssignee(cls, assignee_id):
```
Comment (kobihk) [inline tasks/models.py:38]:
```suggestion
    def filter_by_assignee(cls, assignee_id):
```

File: tasks/models.py [hunk 3]
```diff
@@ -28,6 +28,29 @@ class Task(models.Model):
...
+        if not isinstance(assignee_id, int):
+            raise TypeError
+        try:
+            user = User.objects.get(pk=assignee_id)
+        except User.DoesNotExist:
+            raise ValueError
+        return cls.objects.filter(assignee=user)
+
+    @classmethod
+    def filterByPriority(cls, priority_filter):
```
Comment (kobihk) [inline tasks/models.py:48]:
```suggestion
    def filter_by_priority(cls, priority_filter):
```

File: tasks/test_tasks_filters.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                                    created_by=assigner,
+                                    assignee=assignee,
+                                    status=status,
+                                    priority=priority,
+                                    description=description)
+        task.save()
+        return task
+
+
+@pytest.mark.django_db()
```
Comment (kobihk) [inline tasks/test_tasks_filters.py:72]:
```suggestion
@pytest.mark.django_db
```

File: tasks/test_tasks_filters.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+class TestTasksFilters:
+    """
+    Add test data to database
+    """
+    @pytest.fixture()
```
Comment (kobihk) [inline tasks/test_tasks_filters.py:77]:
```suggestion
    @pytest.fixture
```

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by Asafh7 — DISMISSED** [review]

**Review by kobihk — APPROVED** [review]

**Review by AvivLiberman — APPROVED** [review]
Just need to merge it correctly with my PR #66

**Review by aradz1997 — APPROVED** [review]

**Pull request merged**